ocs_ci/ocs/ocp.py

```diff
@@ -1106,3 +1105,95 @@ def verify_cluster_operator_status(cluster_operator):
     log.info(f'{cluster_operator} status is valid')
 
     return True
+
+
+def validate_cluster_version_status():
+    """
+    Verify OCP upgrade is completed, by checking 'oc get clusterversion'
+    status
+
+    Returns:
+        bool: False in case that one of condition flags is invalid:
+        Progressing (should be False), Failing(should be False)
+        or Available (should be True)
+
+    """
+    ocp = OCP(kind="clusterversion")
+    operator_data = ocp.get('-o json', out_yaml_format=False)
+    conditions = operator_data['items'][0].get('status').get('conditions')
+    for condition in conditions:
+        if condition['type'] == 'Progressing' and condition['status'] == 'True':
+            log.info('cluster version status is Progressing')
+            return False
+        if condition['type'] == 'Failing' and condition['status'] == 'True':
+            log.info('cluster version status is Failing')
+            return False
+        if condition['type'] == 'Available' and condition['status'] != 'True':
+            log.info('cluster status is not available')
+            return False
+
+    log.info('Cluster version validation - OK!')
+    return True
+
+
+def get_ocp_upgrade_channel():
+    """
+    Gets OCP upgrade channel
+
+    Returns:
+        str: OCP upgrade channel name
+
+    """
+    ocp = OCP(kind="clusterversion")
+    log.info("Gathering Subscription Channel information")
+    operator_version = ocp.get('-o json', out_yaml_format=False)
+    log.debug(f"cluster version: {operator_version}")
+    channel = operator_version['items'][0].get('spec').get('channel')
+    log.info(f"Subscription Channel: {channel}")
+
+    return channel
+
+
+def patch_ocp_upgrade_channel(channel_variable):
+    """
+    Using 'oc patch clusterversion' if new OCP upgrade channel is
+    different than current one
+
+    Args:
+        channel_variable (str): New OCP upgrade subscription channel
+
+    """
+    if get_ocp_upgrade_channel() != channel_variable:
+        cmd = (
+            f'patch clusterversions/version -p \'{{"spec":'
+            f'{{"channel":"{channel_variable}"}}}}\' --type=merge'
+        )
+        ocp = OCP()
+        log.info(f"Patching channel into {channel_variable}")
+        ocp.exec_oc_cmd(cmd)
+
+    else:
+        log.info("No patch needed")
+
+
+def verify_ocp_upgrade_channel(channel_variable):
```

Comment:
I think this channel_varibale doesn't have to be as arg at all and you can directly read it in this function from config.
```
 channel_variable = config.UPGRADE['ocp_channel']
```

Right? Or can you think of using this function with other value? If so you can keep it as it is.

Reply:
I would like to leave this function with argument, in case we will need it for other purposes in future, i will add it as default value